### ISSUE TITLE
App not responding or crashing after unmounting OTSession + OTSubscriber

### ISSUE THREAD

**Issue by ngolapnguyen** [opentok/opentok-react-native#366]
# Bug Report
Hi everyone!

I'm building a live-streaming app, with 2 different scenes for Publisher and Subscriber. The Publisher scene has OTSession and OTPublisher. The Subscriber scene has OTSession and OTSubsciber.

The Publisher scene disconnects from the session and unmounts normally, but the Subscriber scene will regularly cause the app to hang (or sometime crash) during unmounting.

- OS: Android
- React Native version: 0.60.5
- opentok-react-native version: 0.12.1
- OpenTok Android SDK version: 2.16.3

**Current bug behavior**
App regularly not responding or crashes after unmounting a scene with OTSession & OTSubscriber.
Sometimes the scene will unmount fine, but only after a long delay.

**Steps to reproduce**
- Navigate to a scene with OTSession & OTSubscriber
- Connect to a session with an ongoing publisher stream
- Go back (unmounting the scene)

**What is the expected *correct* behavior?**
The Subscriber scene should unmount without any issue.

I've been pulling my hair out for the last 3 days trying to find a way to work around this but still in vain. I hope someone can point me in the right direction. I'd really appreciate it!

**Comment by ggoldens** [comment]
I think this is related to the issue #357 
It seems it's an issue on the Android SDK. Will keep you posted. Thanks for reporting!

**Comment by ngo-lap-nguyen** [comment]
It's indeed a SDK-related issue. At the moment I have to downgrade the OpenTok SDK to 2.15.3 as a workaround. The view can unmount without freezing now, but there's still a noticeable delay time until the view is completely unmounted.

**Comment by podvipodvert** [comment]
Some progress on this?

**Comment by enricop89** [comment]
We released today Android SDK `2.16.3` patch. 
Please try to update the version on `android/build.gradle` file and test the issue.

**Comment by podvipodvert** [comment]
there are no any versions on android/build.gradle in "react-native": "0.61.5", and on Ios I using 2.16.3 and I getting an error :
OTSessionManager.disconnectSession was called with 1 arguments but expects 2 arguments. If you haven't changed this method yourself, this usually means that your versions of the native code and JavaScript code are out of sync. Updating both should make this error go away.

**Comment by podvipodvert** [comment]
in your  opentok/opentok-react-native/android/build.gradle  can you change it to 2.16.3

now it like this


dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:${_reactNativeVersion}"  // From node_modules
    implementation 'com.opentok.android:opentok-android-sdk:2.16.1'
}

**Comment by enricop89** [comment]
@podvipodvert:  https://github.com/opentok/opentok-react-native/pull/370/files

**Comment by podvipodvert** [comment]
in branch master, it set as 2.16.1

**Comment by podvipodvert** [comment]
Anyway with 2.16.3 version application not responding after calling this function OT.disconnectSession()

**Comment by QuevedoIB** [comment]
Same problem here: 

React Native version: 0.59.5
opentok-react-native version: 0.11.2
OpenTok Android SDK version: 2.16.3

Freezes and crashes the app on unmounting the OTSession component whenever there's a subscriber to it.

Downgrading the OpenTok Android SDK version: 2.15.3 also causes this problem so the workaround isn't working for me.

Logcat when app crashes:

A/libc: FORTIFY: pthread_mutex_lock called on a destroyed mutex (0x77077a3e98)
A/libc: Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 17015 (mqt_native_modu), pid 16970 

With: 

React Native version: 0.59.5
opentok-react-native version: 0.12.1
OpenTok Android SDK version: 2.16.3

Throws an error for OTSession connect expecting 3 arguments and instead receiving 2.

I really need to fix it so I can give the app to my client and publish it already, any help would be really appreciated.

**Comment by podvipodvert** [comment]
@enricop89 any updates?

**Comment by ggoldens** [comment]
@QuevedoIB that error is because you're using an old version of the opentok-react-native. The current one is 0.12.1. Can you please test updating the library?
Thanks

**Comment by ggoldens** [comment]
Can you please test the branch 0.12.2? If you see it working as expected I'll push it ASAP.

**Comment by andriinegustorov** [comment]
current version 0.12.1 if you released 12.2 so you need to update master branch!

**Comment by ggoldens** [comment]
Now it's released. Please test it again and let us know. Thanks.

**Comment by podvipodvert** [comment]
not  app responding after disconnectSession() but now this error appears: 

OTSessionManager.disconnectSession was called with 1 arguments but expects 2 arguments. If you haven't changed this method yourself, this usually means that your versions of the native code and JavaScript code are out of sync. Updating both should make this error go away.

**Comment by ggoldens** [comment]
Please try to reinstall the library (removing node_modules and then a npm install)
Also, can you try to reproduce it in the basic sample app? https://github.com/opentok/opentok-react-native-samples

**Comment by podvipodvert** [comment]
The same issue in the basic sample app

**Comment by flobady** [comment]
I have the below crash when unmounting the OpenTok component.
"opentok-react-native": "0.12.2"
"react": "16.9.0"
"react-native": "0.61.2"

![Capture d’écran 2019-12-10 à 17 14 36](https://user-images.githubusercontent.com/19803586/70547275-e7f37280-1b70-11ea-86ce-794d1aa78cc9.png)

**Comment by QuevedoIB** [comment]
@ggoldens 

React Native version: 0.59.5
opentok-react-native version: 0.12.2
OpenTok Android SDK version: 2.16.3

On unmounting session throws a Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x30342d6363633c

The tombstone in android logcat is:

<img width="1978" alt="Captura de pantalla 2019-12-12 a las 10 42 51" src="https://user-images.githubusercontent.com/45363951/70701675-0ff7e880-1ccd-11ea-9bf6-872b8f67b187.png">

<img width="1502" alt="Captura de pantalla 2019-12-12 a las 10 44 56" src="https://user-images.githubusercontent.com/45363951/70701711-200fc800-1ccd-11ea-894b-a111ad23889d.png">

I've been struggling for days, any help would be appreciated!

**Comment by capslock35** [comment]
any improvements on this ?

**Comment by andriinegustorov** [comment]
@ggoldens Any updates?

**Comment by ggoldens** [comment]
We're not able to reproduce it. If you are able to reproduce it on the basic sample app, could you please share your App.js file, and provide more info about the device you're testing?
thanks!

**Comment by enricop89** [comment]
Hey guys, can you please let us know if you are able to reproduce it?

**Comment by HarshitPadalia** [comment]
I am facing same issue

**Comment by HarshitPadalia** [comment]
any updated on https://github.com/opentok/opentok-react-native/issues/84?

**Comment by TccTeam** [comment]
my problem was that I used disconnectSession without arguments
OT.disconnectSession(1);

this work for me >>>>
OT.disconnectSession(this.props.sessionId, () =>{});

**Comment by abdulajet** [comment]
Thanks for the issue, can you check out the latest version of the library, and if your issue has not been resolved please reopen/create a new issue.
